core/log-parser/index.js

```diff
@@ -61,6 +66,19 @@ export default class LogParser extends EventEmitter {
     this.linesPerMinute++;
   }
 
+  // manage cleanup disconnected players, session data.
+  clearEventStore() {
+    Logger.verbose('LogParser', 2, 'Cleaning Eventstore');
+    for (const player of Object.values(this.eventStore.players)) {
+      if (this.eventStore.disconnected[player.steamID] === true) {
+        Logger.verbose('LogParser', 2, `Removing ${player.steamID} from eventStore`);
+        delete this.eventStore.players[player.steamID];
+        delete this.eventStore.disconnected[player.steamID];
+      }
+    }
+    this.eventStore.session = {};
```

Comment:
I don't think we currently clean up the event store. Have you tested what side effects this has, e.g. do these events always get triggered before the new game such as someone bleeding out after the end of the game? I'm not fussed about counting the bleed out event, but need to check that kind of thing doesn't cause any errors.

Reply:
https://github.com/Team-Silver-Sphere/SquadJS/blob/17a7c269a7e65013d314bed2f5b56d2a0179409b/squad-server/log-parser/new-game.js#L21

We call the cleanup on newgame, which is triggered on the map change. At this point, all players have been pushed to scoreboard and no more events from damage/kills/wounds are generated. 

Since this cache is self contained within the log parser, it should be self consistent, however, large FTP log parser latencies may cause issues within the server class.

Reply:
The reason for clearing .session every game, is in current/old squadjs, particularly around wounds/damage events, especially when they contain "nullptr" in the regex, is we may be assembling events from data spanning two rounds unintentionally. 

There are alot of edge cases in how damage/wound/died is generated and how we cycle forward attaching information, and I know for a fact this has lead to incorrectly attributed data coming out of the log parser in the current revision. 

While I did beat my head for a while working on a more structured fix, in the end I'm not sure how to deal with ambiguous cases and so left this as is for now.

Reply:
> At this point, all players have been pushed to scoreboard and no more events from damage/kills/wounds are generated.
Is this something that's changed? I thought previous tickets were still updated when the scoreboard was loaded as people were bleeding out during that time, etc.

Reply:
As for why I keep disconnected and players separate and only run the cleanup on new game:

My concern was probably what your pointing to: If we have a bleedout after a player leaves, we won't have the disconnected player within the cache.

The solution was to *keep* the players within the cache until the round ends. 

So, in practice,A new round starts:

Player A downs Player B
Player A Disconnects. 

Player B bleeds out (we attach player A).

New game fires, and we cleanup player A.

Reply:
> Is this something that's changed? I thought previous tickets were still updated when the scoreboard was loaded as people were bleeding out during that time, etc.

So, when we hit the scoreboard, all players "unposses" their objects, or should from testing. 

```
[2022.12.29-09.08.22:442][813]LogGameMode: Display: Match State Changed from InProgress to WaitingPostMatch
[2022.12.29-09.08.22:442][813]LogSquadTrace: [DedicatedServer]OnUnPossess(): PC=Doug1952
[2022.12.29-09.08.22:442][813]LogSquadTrace: [DedicatedServer]OnUnPossess(): PC=Doug1952 current health value 5.000000
[2022.12.29-09.08.22:442][813]LogSquadTrace: [DedicatedServer]ChangeState(): PC=Doug1952 OldState=Playing NewState=Inactive
[2022.12.29-09.08.22:442][813]LogSquadTrace: [DedicatedServer]BeginInactiveState(): PC=Doug1952
[2022.12.29-09.08.22:443][813]LogSquadTrace: [DedicatedServer]OnUnPossess(): PC=Tmr539
[2022.12.29-09.08.22:443][813]LogSquadTrace: [DedicatedServer]OnUnPossess(): PC=Tmr539 current health value -9.223495
[2022.12.29-09.08.22:443][813]LogSquadTrace: [DedicatedServer]ChangeState(): PC=Tmr539 OldState=Playing NewState=Inactive
[2022.12.29-09.08.22:443][813]LogSquadTrace: [DedicatedServer]BeginInactiveState(): PC=Tmr539
[2022.12.29-09.08.22:444][813]LogSquadTrace: [DedicatedServer]OnUnPossess(): PC=Slipped DabSloth
[2022.12.29-09.08.22:444][813]LogSquadTrace: [DedicatedServer]OnUnPossess(): PC=Slipped DabSloth current health value 35.220371
[2022.12.29-09.08.22:444][813]LogSquadTrace: [DedicatedServer]ChangeState(): PC=Slipped DabSloth OldState=Playing NewState=Inactive
[2022.12.29-09.08.22:444][813]LogSquadTrace: [DedicatedServer]BeginInactiveState(): PC=Slipped DabSloth
```

Reply:
I presume once unpossessed they are cleaned up, or at least their deaths, etc. aren't logged?

Reply:
They don't count as "deaths" from the perspective of player-died.

IE, we won't see the regex for player died between the above log and the next newgame.

Reply:
In testing, the above is 100% the case, 

```
PLAYER_UNPOSSESS 2022-12-29T09:08:22.442Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.442Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.443Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.443Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.444Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.444Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.445Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.445Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.446Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.447Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.447Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.448Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.448Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.449Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.449Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.450Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.450Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.451Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.453Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.453Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.454Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.454Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.455Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.455Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.456Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.456Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.457Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.457Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.458Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.458Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.458Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.458Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.459Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.459Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.460Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.460Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.461Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.461Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.462Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.462Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.463Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.463Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.463Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.463Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.464Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.464Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.465Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.465Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.466Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.466Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.467Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.467Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.468Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.468Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.469Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.469Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.470Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.470Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.471Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.471Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.471Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.471Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.472Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.472Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.473Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.473Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.474Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.474Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.475Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.475Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.476Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.476Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.477Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.477Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.477Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.477Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.478Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.478Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.479Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.479Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.480Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.480Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.481Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.482Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.485Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.485Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.485Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.485Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.486Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.486Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.487Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.487Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.488Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.488Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.489Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.489Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.490Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.490Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.491Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.491Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.492Z
PLAYER_UNPOSSESS 2022-12-29T09:08:22.492Z
PLAYER_UNPOSSESS 2022-12-29T09:08:26.921Z
PLAYER_UNPOSSESS 2022-12-29T09:08:34.987Z
PLAYER_UNPOSSESS 2022-12-29T09:08:37.683Z
PLAYER_UNPOSSESS 2022-12-29T09:08:39.354Z
PLAYER_UNPOSSESS 2022-12-29T09:08:39.421Z
PLAYER_UNPOSSESS 2022-12-29T09:08:40.691Z
PLAYER_UNPOSSESS 2022-12-29T09:08:41.448Z
PLAYER_UNPOSSESS 2022-12-29T09:08:43.164Z
PLAYER_UNPOSSESS 2022-12-29T09:08:44.166Z
PLAYER_UNPOSSESS 2022-12-29T09:08:45.214Z
PLAYER_UNPOSSESS 2022-12-29T09:08:45.682Z
PLAYER_UNPOSSESS 2022-12-29T09:08:47.152Z
PLAYER_UNPOSSESS 2022-12-29T09:08:49.092Z
NEW_GAME Fallujah_RAAS_v7 2022-12-29T09:08:54.391Z```

Excuse the formatting, this was done offline

Reply:
[out.zip](https://github.com/Team-Silver-Sphere/SquadJS/files/10329688/out.zip)

Reply:
On tickets, it may be the case that players that have negative health values (but not -300) effect tickets when they are cleaned up as above, but this is an issue I'm not sure we can work around. 

This *could* by why sometimes we see tickets adjusting at the scoreboard, but I'm not entirely sure. I think players leaving also may have an effect here, but these are squad bugs, not squadjs bugs. 

As for renaming the PR, I can do that. 

Not sure whats appropriate since this rolls a few things together;